oak_idl/src/lib.rs

```diff
@@ -24,70 +24,117 @@ use core::fmt::Debug;
 
 pub mod utils;
 
-// TODO(#2500): Align with gRPC Status.
-// Ref: https://github.com/grpc/grpc/blob/master/src/proto/grpc/status/status.proto
-/// Logical error model.
 #[derive(Debug)]
-pub struct Error {
-    pub code: ErrorCode,
+pub struct Status {
+    pub code: StatusCode,
     /// English message that helps developers understand and resolve the error.
     pub message: String,
 }
 
-impl Error {
-    pub fn new(code: ErrorCode) -> Self {
+impl Status {
+    pub fn new(code: StatusCode) -> Self {
         Self {
             code,
             message: "".to_string(),
         }
     }
 
-    pub fn new_with_message(code: ErrorCode, message: impl Into<String>) -> Self {
+    pub fn new_with_message(code: StatusCode, message: impl Into<String>) -> Self {
         Self {
             code,
             message: message.into(),
         }
     }
 }
 
-// TODO(#2500): Align these with gRPC status codes.
-/// Basic error code to categorize the error. Error codes are represented as an
-/// u32 id. The id `0` is reserved, as it may be used to indicate an "Ok" status
-/// in transport implementations.
```

Comment:
This led to a somewhat messy implementation where the transport layers handles some status codes (but not others), which @conradgrobler rightly pointed out wasn't great: https://github.com/project-oak/oak/pull/2944#discussion_r893102044